homeassistant/components/icloud/account.py

```diff
@@ -124,17 +124,23 @@ def setup(self) -> None:
                 ),
                 self._config_entry.data[CONF_USERNAME],
             )
-
-            self.hass.add_job(
-                self.hass.config_entries.flow.async_init(
-                    DOMAIN,
-                    context={"source": SOURCE_REAUTH},
-                    data={
-                        **self._config_entry.data,
-                        "unique_id": self._config_entry.unique_id,
-                    },
+            context = {
+                "source": SOURCE_REAUTH,
+                "unique_id": self._config_entry.unique_id,
+            }
+            if not any(
+                flow
+                for flow in self.hass.config_entries.flow.async_progress()
```

Comment:
We seem to be in sync context so we can't use the async api here.

Reply:
actually, that brings up a question. The function is called async, but it's a synchronous callback, presumably because it's returning values from a dict. Do we need to treat it as calling an async function from a sync context?